src/GraphQL/loaders/classSchemaTypes.js

```diff
@@ -0,0 +1,448 @@
+  fields: {
+    name: SCHEMA_FIELD_NAME_ATT,
+  },
+});
+
+const SCHEMA_POLYGON_FIELD = new GraphQLObjectType({
+  name: 'SchemaPolygonField',
+  description:
+    'The SchemaPolygonField is used to return information of a Polygon field.',
+  interfaces: [SCHEMA_FIELD],
+  fields: {
+    name: SCHEMA_FIELD_NAME_ATT,
+  },
+});
+
+const SCHEMA_BYTES_FIELD_INPUT = new GraphQLInputObjectType({
+  name: 'SchemaBytesFieldInput',
+  description:
+    'The SchemaBytesFieldInput is used to specify a field of type bytes for an object class schema.',
+  fields: {
+    name: SCHEMA_FIELD_NAME_ATT,
+  },
+});
+
+const SCHEMA_BYTES_FIELD = new GraphQLObjectType({
+  name: 'SchemaBytesField',
+  description:
+    'The SchemaBytesField is used to return information of a Bytes field.',
+  interfaces: [SCHEMA_FIELD],
+  fields: {
+    name: SCHEMA_FIELD_NAME_ATT,
+  },
+});
+
+const TARGET_CLASS_ATT = {
+  description: 'This is the name of the target class for the field.',
+  type: new GraphQLNonNull(GraphQLString),
+};
+
+const SCHEMA_POINTER_FIELD_INPUT = new GraphQLInputObjectType({
+  name: 'PointerFieldInput',
+  description:
+    'The PointerFieldInput is used to specify a field of type pointer for an object class schema.',
+  fields: {
+    name: SCHEMA_FIELD_NAME_ATT,
+    targetClassName: TARGET_CLASS_ATT,
+  },
+});
+
+const SCHEMA_POINTER_FIELD = new GraphQLObjectType({
+  name: 'SchemaPointerField',
+  description:
+    'The SchemaPointerField is used to return information of a Pointer field.',
+  interfaces: [SCHEMA_FIELD],
+  fields: {
+    name: SCHEMA_FIELD_NAME_ATT,
+    targetClassName: TARGET_CLASS_ATT,
+  },
+});
+
+const SCHEMA_RELATION_FIELD_INPUT = new GraphQLInputObjectType({
+  name: 'RelationFieldInput',
+  description:
+    'The RelationFieldInput is used to specify a field of type relation for an object class schema.',
+  fields: {
+    name: SCHEMA_FIELD_NAME_ATT,
+    targetClassName: TARGET_CLASS_ATT,
+  },
+});
+
+const SCHEMA_RELATION_FIELD = new GraphQLObjectType({
+  name: 'SchemaRelationField',
+  description:
+    'The SchemaRelationField is used to return information of a Relation field.',
+  interfaces: [SCHEMA_FIELD],
+  fields: {
+    name: SCHEMA_FIELD_NAME_ATT,
+    targetClassName: TARGET_CLASS_ATT,
+  },
+});
+
+const SCHEMA_ACL_FIELD = new GraphQLObjectType({
+  name: 'SchemaACLField',
+  description:
+    'The SchemaACLField is used to return information of an ACL field.',
+  interfaces: [SCHEMA_FIELD],
+  fields: {
+    name: SCHEMA_FIELD_NAME_ATT,
+  },
+});
+
+const SCHEMA_FIELDS_INPUT = new GraphQLInputObjectType({
+  name: 'SchemaFieldsInput',
+  description: `The CreateClassSchemaInput type is used to specify the schema for a new object class to be created.`,
+  fields: {
+    addStrings: {
+      description:
+        'These are the String fields to be added to the class schema.',
+      type: new GraphQLList(new GraphQLNonNull(SCHEMA_STRING_FIELD_INPUT)),
```

Comment:
Why do we need one input type for each data type? Couldn't this input type be more generic?

Reply:
As we talked, two reasons:
- Pointer and Relation requires a different input, because they also need the targetClassName to be defined;
- In the future we will add the defaultValue input, and it will be different for each field type.